### PR TITLE
Feature: Assets Improvements

### PR DESCRIPTION
**Pull request by Zyie**
This PR aims to make using Assets a little more powerful and consistent

### Type Changes

- To simplify things everything is either a `ResolvedAsset` or `UnresolvedAsset`
- I've deprecated `name` and `srcs` in favour of `alias` and `src` to be consistent between the array and object versions
- Renamed `ResolverBundle` to `AssetsBundle`
- Renamed `ResolverManifest` to `AssetsManifest`
- Renamed `ResolverAssetsObject` to `UnresolvedAssetObject`
- Deprecated `ResolverAssetsArray` for nothing as `UnresolvedAsset[]` is the same thing
- Deprecated `ResolveAsset` and `LoadAsset` as these are now `ResolvedAsset`

### Manifest Changes

- Manifest asset arrays and objects now accept the same object (minus the name as it uses the key for that)
e.g. 
```js
const manifestObject = {
  bundles: [
    {
      name: "initialLoad",
      assets: {
        defaultFont: {
          src: "./fonts/arial/ArialRoundedMTBold.{woff2,woff,ttf}",
        },
      },
    },
  ],
};
const manifestArray = {
  bundles: [
    {
      name: "initialLoad",
      assets: [
        {
          alias: "defaultFont",
          src: "./fonts/arial/ArialRoundedMTBold.{woff2,woff}",
        },
      ],
    },
  ],
};
```
- Previously it was not possible for the object version to accept an array of sources, this has been fixed. Fixes commented issue in #9251
- `loadParser` now works with the manifest

### API Changes

- `Assets.add` now accepts an object instead of individual arguments, allows for data to be passed into individual sources, and accepts a much wider range of sources to be consistent with how manifests work
e.g.
```js
// Old
Assets.add('defaultFont', './fonts/arial/ArialRoundedMTBold.{woff2,woff}', { type: 'font' });

// New
Assets.add({
    alias: 'defaultFont',
    src: './fonts/arial/ArialRoundedMTBold.{woff2,woff}',
    data: { type: 'font' },
});

// Previously this was not possible
Assets.add({
    alias: 'defaultFont',
    src: './fonts/arial/ArialRoundedMTBold.{woff2,woff}',
    data: { type: 'font' },
});
Assets.add({
    alias: 'defaultFont',
    src: [
      {
        src: './fonts/arial/ArialRoundedMTBold.woff',
        data: { type: 'font' },
        loadParser: 'loadWebfont',
      },
      {
        src: './fonts/arial/ArialRoundedMTBold.woff2',
        data: { type: 'different Font' },
        loadParser: 'loadWithDifferentParser',
      },
    ],
    data: { type: 'font' },
});
```

- `Assets.load` now also allows you to load urls with glob patterns. Fixes main issue in #9251
e.g.
```js
// Old
Assets.add('defaultFont', 'defaultFont.{woff,woff2}');
Assets.load('defaultFont');

// New
Assets.load('defaultFont.{woff,woff2}');
Assets.load({
    alias: 'defaultFont',
    src: 'defaultFont.{woff,woff2}',
    data: { type: 'font' },
});
```

### CODE REVIEW

**Comment by GoodBoyDigital** [comment]
Yoo! Thanks for the PR, this is cool. Just wanted to clarify that `Old` still works in this instance?

```

// Old
Assets.add('defaultFont', 'defaultFont.{woff,woff2}');
Assets.load('defaultFont');

// New
Assets.load('defaultFont.{woff,woff2}');
Assets.load({
    alias: 'defaultFont',
    src: 'defaultFont.{woff,woff2}',
    data: { type: 'font' },
});

```

**Comment by Zyie** [comment]
> Yoo! Thanks for the PR, this is cool. Just wanted to clarify that `Old` still works in this instance?
> 
> ```
> 
> // Old
> Assets.add('defaultFont', 'defaultFont.{woff,woff2}');
> Assets.load('defaultFont');
> ```
This does still work. We even have tests for it!

File: packages/assets/src/types-deprecated.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import type { AssetsBundle, AssetsManifest, ResolvedAsset, UnresolvedAsset, UnresolvedAssetObject } from './types';
+
+/**
+ * Please use `ResolvedAsset` instead.
+ * @memberof PIXI
+ * @deprecated since 7.2.0
```
Comment (bigtimebuddy) [inline packages/assets/src/types-deprecated.ts:6]: Can you change these versions to 7.3.0?

File: packages/assets/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ * A fully resolved asset, with all the information needed to load it.
+ * @memberof PIXI
+ */
+export interface ResolvedAsset<T=any>
+{
+    /** Aliases associated with asset */
+    alias?: string[];
+    /**
+     * Please use `alias` instead.
+     * @deprecated since 7.2.0
```
Comment (bigtimebuddy) [inline packages/assets/src/types.ts:19]: Same here, 7.3.0

**Review by bigtimebuddy — APPROVED** [review]
Minor doc things, but otherwise, I like the name changes. Much clearer.

**Pull request merged**